pkg/archive/archive_test.go

```diff
@@ -207,6 +207,164 @@ func TestCmdStreamGood(t *testing.T) {
 	}
 }
 
+func TestUntarPathWithInvalidSrc(t *testing.T) {
+	_, err := os.Create("/tmp/dest")
+	if err != nil {
+		t.Fatalf("Fail to create the destination file")
+	}
+	defer os.RemoveAll("/tmp/dest")
+	err = UntarPath("", "/tmp/dest")
+	if err == nil {
+		t.Fatalf("UntarPath with invalid src path should throw an error.")
+	}
+}
+
+func TestUntarPath(t *testing.T) {
+	cmd := exec.Command("/bin/sh", "-c", "touch /tmp/src && tar cfT /tmp/src.tar /tmp/src")
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll("/tmp/src/")
+	defer os.RemoveAll("/tmp/src.tar")
+	_, err = os.Create("/tmp/dest")
+	if err != nil {
+		t.Fatalf("Fail to create the destination file")
+	}
+	defer os.RemoveAll("/tmp/dest")
+	err = UntarPath("/tmp/src.tar", "/tmp/dest")
+	if err != nil {
+		t.Fatalf("UntarPath shouldn't throw an error, %s.", err)
+	}
+	_, err = os.Stat("/tmp/src")
+	if err != nil {
```

Comment:
Not sure what you're testing here.  You created /tmp/src via the touch, you don't delete it (except via a defer for later processing), so why would this test ever fail? And what did this have to do with the untar?  Did you mean to test for /tmp/dest/src instead? If so, then you'll need to fix your tar because the tar file is empty based on how you created it.